packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,328 @@
+    `;
+
+    const labelStyle = css`
+      color: #3d4f58;
+      font-size: 14px;
+      font-weight: bold;
+      height: 20px;
+      padding-bottom: 8px;
+    `;
+
+    const descriptionStyle = css`
+      color: #5d6c74;
+      font-size: 14px;
+      height: ${description === '' ? '0' : '20'}px;
+      font-weight: normal;
+      padding-bottom: 8px;
+    `;
+
+    const inputContainerStyle = css`
+      position: relative;
+      display: flex;
+      align-items: center;
+      z-index: 0;
+    `;
+
+    const inputStyle = css`
+      width: 400px;
+      height: 36px;
+      border-radius: 4px;
+      padding-left: 12px;
+      font-size: 14px;
+      font-weight: normal;
+      &:placeholder {
+        color: #89989b;
+      }
+      &:focus {
+        outline: none;
+      }
+    `;
+
+    const errorIconStyle = css`
+      position: absolute;
+      right: 10px;
+      color: #cf4a22;
+    `;
+
+    const validIconStyle = css`
+      position: absolute;
+      right: 10px;
+      color: #13aa52;
+    `;
+
+    const optionalStyle = css`
+      position: absolute;
+      right: 12px;
+      color: #5d6c74;
+      font-size: 12px;
+      font-style: italic;
+      font-weight: normal;
+    `;
+
+    const errorMessageStyle = css`
+      color: #cf4a22;
+      font-size: 14px;
+      height: 20px;
+      padding-top: 4px;
+      font-weight: normal;
+    `;
+
+    return (
+      <div className={className}>
+        <label className={textInputStyle + ' ' + labelStyle}>
+          {label}
+          <label className={descriptionStyle}>{description}</label>
+          <div className={inputContainerStyle}>
+            <input
+              {...inputSelectorProp.prop}
+              className={cx(
+                inputStyle,
+                css`
+                  z-index: ${hasFocus ? 2 : 1};
+                  background: ${disabled ? '#E7EEEC' : '#FFFFFF'};
+                  padding-right: ${getInputPaddingFromState()};
+                  border: 1px solid ${getInputColorFromState()};
+                `,
+              )}
+              type="text"
+              value={value}
+              required={!optional}
+              disabled={disabled}
+              placeholder={placeholder}
+              onChange={e => onValueChange(e)}
+              ref={forwardRef}
+              onFocus={() => setHasFocus(true)}
+              onBlur={() => setHasFocus(false)}
+            />
+            {state === State.valid && (
+              <Icon
+                glyph="Checkmark"
+                className={cx(
+                  validIconStyle,
+                  css`
+                    z-index: ${hasFocus ? 2 : 1};
+                  `,
+                )}
+              />
+            )}
+            {state === State.error && (
+              <Icon
+                glyph="Warning"
+                className={cx(
+                  errorIconStyle,
+                  css`
+                    z-index: ${hasFocus ? 2 : 1};
```

Comment:
Question: What's the reason this is needed?

Reply:
Without this line, the interaction ring always renders on top of the input element. I don't know that I could explain why this particular CSS fixes the problem, it was just something I found in Brooke's wrapper code.